engine/gamesys/src/gamesys/components/comp_model.cpp

```diff
@@ -1062,24 +1066,27 @@ namespace dmGameSystem
             #define UNPACK_ATTRIBUTE_PTR(name) \
                 (instance_render_item->m_Mesh->name.m_Count ? instance_render_item->m_Mesh->name.m_Data : 0)
 
-                const float* uv_channels[] = {
-                    UNPACK_ATTRIBUTE_PTR(m_Texcoord0),
-                    UNPACK_ATTRIBUTE_PTR(m_Texcoord1),
-                };
+                const float* world_matrix_channels[]  = { (float*) &instance_render_item->m_World };
+                const float* normal_matrix_channels[] = { (float*) &normal_matrix };
+                const float* uv_channels[]            = { UNPACK_ATTRIBUTE_PTR(m_Texcoord0), UNPACK_ATTRIBUTE_PTR(m_Texcoord1), };
+                const float* color_channels[]         = { UNPACK_ATTRIBUTE_PTR(m_Colors) };
+                const float* position_channels[]      = { UNPACK_ATTRIBUTE_PTR(m_Positions) };
+                const float* normal_channels[]        = { UNPACK_ATTRIBUTE_PTR(m_Normals) };
+                const float* tangent_channels[]       = { UNPACK_ATTRIBUTE_PTR(m_Tangents) };
```

Comment:
The source data has the exact same layout now for all data points, which should make it easier to introduce more channels if needed.